api/tests/routes/backoffice/admin_test.py

```diff
@@ -103,6 +106,76 @@ def test_update_role_with_empty_permissions(self, authenticated_client):
 
+    def test_log_update_role(self, legit_user, authenticated_client):
+        permission1 = perm_factories.PermissionFactory(name="Gérer les droits")
+        permission2 = perm_factories.PermissionFactory(name="Visualiser les comptes admin")
+        permission3 = perm_factories.PermissionFactory(name="Activer/désactiver les feature flags")
+        perms = [permission1, permission2, permission3]
+        perm_factories.PermissionFactory(name="Suspendre un compte jeune")
+        old_perms = [permission1, permission2]
+        role_to_edit = perm_factories.RoleFactory(
+            name="SUPPORT_N1",
+            permissions=old_perms,
+        )
+        role_not_to_edit = perm_factories.RoleFactory(permissions=old_perms)
+
+        new_perms = [permission2, permission3]
+        base_form = {}
+        for perm in perms:
+            base_form[perm.name] = perm in new_perms
+
+        response = self.post_to_endpoint(authenticated_client, role_id=role_to_edit.id, form=base_form)
+        assert response.status_code == 303
+
+        action = history_models.ActionHistory.query.one()
+        assert action.actionType == history_models.ActionType.ROLE_PERMISSIONS_CHANGED
+        assert action.authorUserId == legit_user.id
+        assert action.extraData == {
+            "role_name": "support_n1",
+            "modified_info": {
+                role_not_to_edit.permissions[0].name: {"new_info": False, "old_info": True},
+                role_to_edit.permissions[1].name: {"new_info": True, "old_info": False},
+            },
+        }
+
+
+class GetRolesHistoryTest(GetEndpointHelper):
+    endpoint = "backoffice_web.get_roles_history"
+    needed_permission = perm_models.Permissions.MANAGE_PERMISSIONS
+
+    def test_get_log_history_admin(self, legit_user, authenticated_client):
+        permission1 = perm_factories.PermissionFactory(name="Gérer les droits")
+        permission2 = perm_factories.PermissionFactory(name="Visualiser les comptes admin")
+        perm_factories.PermissionFactory(name="Activer/désactiver les feature flags")
+        perm_factories.PermissionFactory(name="Suspendre un compte jeune")
+
+        old_perms = [permission1, permission2]
+        role_to_edit = perm_factories.RoleFactory(name="SUPPORT_N1", permissions=old_perms)
+        role_not_to_edit = perm_factories.RoleFactory(permissions=old_perms)
+        action = history_factories.ActionHistoryFactory(
+            actionType=history_models.ActionType.ROLE_PERMISSIONS_CHANGED,
+            authorUser=legit_user,
+            comment=None,
+            extraData={
+                "role_name": role_to_edit.name,
+                "modified_info": {
+                    role_not_to_edit.permissions[0].name: {"new_info": False, "old_info": True},
+                    role_not_to_edit.permissions[1].name: {"new_info": True, "old_info": False},
+                },
+            },
+        )
+
+        response = authenticated_client.get(url_for(self.endpoint))
```

Comment:
Il est important de vérifier le code retour des requêtes HTTP.
De plus, j'avais fait en sorte de tester le nombre de requêtes SQL de toutes les requêtes HTTP GET, donc je souhaiterais qu'on ne l'oublie pas dans les nouvelles.
```suggestion
        with assert_num_queries(self.expected_num_queries):
            response = authenticated_client.get(url_for(self.endpoint))
            assert response.status_code == 200
```
A priori, tu devrais définir `expected_num_queries = 3` pour cette classe ; sinon on a des requêtes SQL en trop !